examples/CREATE_AND_USE_A_SESSION.md

```diff
@@ -0,0 +1,86 @@
+### Create and Use a Session
+
+| Key                                                                                         | Description           |
+| ------------------------------------------------------------------------------------------- | --------------------- |
+| [sessionConfigs](https://bcnmy.github.io/biconomy-client-sdk/types/BuildUserOpOptions.html) | CreateSessionConfig[] |
+
+```typescript
+import {
+  createSmartAccountClient,
+  createSession,
+  createSessionSmartAccountClient,
+} from "@biconomy/account";
+import { createWalletClient, http, createPublicClient } from "viem";
+import { privateKeyToAccount, generatePrivateKey } from "viem/accounts";
+import { mainnet as chain } from "viem/chains";
+
+const account = privateKeyToAccount(generatePrivateKey());
+const signer = createWalletClient({ account, chain, transport: http() });
+const smartAccount = await createSmartAccountClient({
+  signer,
+  bundlerUrl,
+  paymasterUrl,
+}); // Retrieve bundler and pymaster urls from dashboard
+const smartAccountAddress = await smartAccount.getAccountAddress();
+
+const { sessionKeyAddress, sessionStorageClient } =
+  await createAndStoreNewSessionKey(smartAccount, chain);
+
+const { wait, session } = await createSession(
+  smartAccount,
+  sessionKeyAddress,
+  [
+    {
+      sessionKeyAddress,
+      contractAddress: nftAddress,
+      functionSelector: "safeMint(address)",
+      rules: [
+        {
+          offset: 0,
+          condition: 0,
+          referenceValue: pad(smartAccountAddress, { size: 32 }),
+        },
+      ],
+      interval: {
+        validUntil: 0,
+        validAfter: 0,
+      },
+      valueLimit: 0n,
+    },
+  ],
+  sessionStorageClient,
+  {
+    paymasterServiceData: { mode: PaymasterMode.SPONSORED },
+  }
+);
+
+const {
+  receipt: { transactionHash },
+} = await wait();
+
+const smartAccountWithSession = await createSessionSmartAccountClient(
```

Comment:
so I would use createSessionSmartAccountClient or createSmartAccountClient?

createSessionSmartAccountClient gives me smart account with active validation module set as session key manager?

Reply:
Yes that's right. They both return the same thing (BiconomySmartAccountV2) but there are subtle differences in the configuration. `createSessionSmartAccountClient` expects *no* signer and a mandatory smartAccountAddress in the config (so a dapp can use the session without the original signer). It will also pass along the module info to be included with each request that it makes so that additional module-specific arguments do not need to be passed while sending transactions for example.

Reply:
I see. do you not need to pass the sessionSigner object?

Reply:
That's reconstituted from the session param (the second param that's passed into this function)

Reply:
I see